extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/CdcSinks.java

```diff
@@ -0,0 +1,270 @@
+        return sink(name, map, clientConfig, keyFn, valueFn);
+    }
+
+    @Nonnull
+    private static <K, V> Sink<ChangeRecord> sink(
+            @Nonnull String name,
+            @Nonnull String map,
+            @Nullable ClientConfig clientConfig,
+            @Nonnull FunctionEx<ChangeRecord, K> keyFn,
+            @Nonnull FunctionEx<ChangeRecord, V> valueFn
+    ) {
+        ProcessorSupplier supplier = AbstractHazelcastConnectorSupplier.of(asXmlString(clientConfig),
+                instance -> new CdcSinkProcessor<>(instance, map, keyFn, extend(valueFn)));
+        ProcessorMetaSupplier metaSupplier = ProcessorMetaSupplier.forceTotalParallelismOne(supplier, name);
+        return new SinkImpl<>(name, metaSupplier, true, null);
+    }
+
+    @Nonnull
+    private static <V> FunctionEx<ChangeRecord, V> extend(@Nonnull FunctionEx<ChangeRecord, V> valueFn) {
+        return (record) -> {
+            if (DELETE.equals(record.operation())) {
+                return null;
+            }
+            return valueFn.apply(record);
+        };
+    }
+
+    private static class CdcSinkProcessor<K, V> extends UpdateMapWithMaterializedValuesP<ChangeRecord, K, V> {
+
+        private Sequences<K> sequences;
+
+        CdcSinkProcessor(
+                @Nonnull HazelcastInstance instance,
+                @Nonnull String map,
+                @Nonnull FunctionEx<? super ChangeRecord, ? extends K> keyFn,
+                @Nonnull FunctionEx<? super ChangeRecord, ? extends V> valueFn
+        ) {
+            super(instance, map, keyFn, valueFn);
+        }
+
+        @Override
+        public void init(@Nonnull Outbox outbox, @Nonnull Context context) {
+            super.init(outbox, context);
+
+            HazelcastProperties properties = new HazelcastProperties(context.jetInstance().getConfig().getProperties());
+            int expiration = properties.getSeconds(SEQUENCE_CACHE_EXPIRATION_SECONDS);
+            this.sequences = new Sequences<>(expiration);
+        }
+
+        @Override
+        protected boolean shouldBeDropped(K key, ChangeRecord item) {
+            long timestamp = getTimestamp(item);
+
+            ChangeRecordImpl recordImpl = (ChangeRecordImpl) item;
+            long sequencePartition = recordImpl.getSequencePartition();
+            long sequenceValue = recordImpl.getSequenceValue();
+
+            boolean isNew = sequences.update(key, timestamp, sequencePartition, sequenceValue);
+            return !isNew;
+        }
+
+        private static long getTimestamp(ChangeRecord item) {
+            try {
+                return item.timestamp();
+            } catch (ParsingException e) {
+                //use current time, should be good enough for cache expiration purposes
+                return System.currentTimeMillis();
+            }
+        }
+
+    }
+
+    private static final class Sequences<K> {
+
+        private static final int INITIAL_CAPACITY = 64 * 1024;
```

Comment:
this is quite large

Reply:
I've set a large value due to what's in the Javadoc, but I might have overdone it, will decrease it.

> A linked hash map has two parameters that affect its performance: initial capacity and load factor. They are defined precisely as for HashMap. Note, however, that the penalty for choosing an excessively high value for initial capacity is less severe for this class than for HashMap, as iteration times for this class are unaffected by capacity.